### PR TITLE
add detailed error config (#71)

### PR DESCRIPTION
**Pull request by muhammedbehram**
Created a setting key as mentioned in issue #71 and passed it into tasks.

### CODE REVIEW

File: plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala [hunk 1]
```diff
@@ -100,11 +102,12 @@ object ScalafmtPlugin extends AutoPlugin {
       config: Path,
       log: Logger,
       writer: OutputStreamWriter,
-      resolvers: Seq[Resolver]
+      resolvers: Seq[Resolver],
+      isDetailedError: Boolean
```
Comment (poslegm) [inline plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala:106]:
```suggestion
      detailedErrorEnabled: Boolean
```

more understandable name

File: plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala [hunk 2]
```diff
@@ -62,6 +62,8 @@ object ScalafmtPlugin extends AutoPlugin {
       "Execute the scalafmtCheck task for all configurations in which it is enabled. " +
         "(By default this means the Compile and Test configurations.)"
     )
+    val scalafmtDetailedError =
+      settingKey[Boolean]("Log detailed error with stack trace, off by default")
```
Comment (poslegm) [inline plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala:66]: Not sure about English grammar :)

May be `Enables logging of detailed errors with stacktraces, disabled by default`

**Review by poslegm — CHANGES_REQUESTED** [review]
Thank you! LGTM with small remarks

File: plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala [hunk 3]
```diff
@@ -323,14 +358,15 @@ object ScalafmtPlugin extends AutoPlugin {
   }
 
   private def scalafmtTask =
-    Def.task {
+    Def.task { //TODO implement from here
```
[the remark this answers is not in the document]
Comment (muhammedbehram) [reply plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala:361]: It was a reminder for myself. Sorry that I failed to check it before sending the commit. I'll remove it and send it with other changes.

File: plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala (hunk 2 above)
Comment (muhammedbehram) [reply plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala:66]: It sounds better :)

**Comment by poslegm** [comment]
Compilation errors: 

```
[error] /home/runner/work/sbt-scalafmt/sbt-scalafmt/plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala:432:17: The key `scalafmtTask` is not being invoked inside the task definition.
[error] 
[error] problem:  Keys missing `.value` are not initialized and their dependency is not registered.
[error] solution: Replace `scalafmtTask` by `scalafmtTask.value` or remove it if unused.
[error]     
[error]     scalafmt := scalafmtTask,
```

You can test plugin locally with `sbt publishLocal` or run unit tests with `sbt plugin/scripted`

**Comment by muhammedbehram** [comment]
> You can test plugin locally with `sbt publishLocal` or run unit tests with `sbt plugin/scripted`

After I fixed the complication issue,
I tried `sbt publishLocal` both on master branch and `add-detailed-error-config` branch. It failed with the same error.
`[error]  Cause of test exception: {line 83}  Command succeeded but failure was expected` 
This issue seems to be a local one but I thought it would be for me to report this before sending the fix commit for the compilation error.

**Comment by poslegm** [comment]
> This issue seems to be a local one but I thought it would be better for me to report this before sending the fix commit for the compilation error.

It's strange. Looks really like problem in local environment because I built plugin successfully just now.

**Comment by muhammedbehram** [comment]
> It's strange. Looks really like problem in local environment because I built plugin successfully just now.

I couldn't find a solution yet. Still committed changes. Would you like me to squash commits?

**Comment by poslegm** [comment]
I tested it locally, everything is fine 👍🏻

**Comment by poslegm** [comment]
You can add flag description [here](https://scalameta.org/scalafmt/docs/installation.html#sbt) after v2.4.3 will be released

**Pull request merged**